src/components/tl/ucp/tl_ucp.c

```diff
@@ -38,6 +38,21 @@ static ucc_config_field_t ucc_tl_ucp_lib_config_table[] = {
      ucc_offsetof(ucc_tl_ucp_lib_config_t, allreduce_kn_radix),
      UCC_CONFIG_TYPE_UINT},
 
+    {"ALLREDUCE_SRA_KN_RADIX", "4",
+     "Radix of the scatter-reduce-allgather (SRA) knomial allreduce algorithm",
+     ucc_offsetof(ucc_tl_ucp_lib_config_t, allreduce_sra_kn_radix),
+     UCC_CONFIG_TYPE_UINT},
+
+    {"REDUCE_SCATTER_KN_RADIX", "4",
+     "Radix of the knomial reduce-scatter algorithm",
+     ucc_offsetof(ucc_tl_ucp_lib_config_t, reduce_scatter_kn_radix),
+     UCC_CONFIG_TYPE_UINT},
+
+    {"ALLGATHER_KN_RADIX", "4",
+     "Radix of the knomial allgather algorithm",
+     ucc_offsetof(ucc_tl_ucp_lib_config_t, allgather_kn_radix),
+     UCC_CONFIG_TYPE_UINT},
+
```

Comment:
What do you mean - single var for all the radixes? We usually use different radixes for small/large msg (small ar, and sra e.g.). I think. Do you mean adding separate table for "radix" variables which will inherit from the 1 common table? So that we could set them all?

Reply:
Not a separate table, but something like UCC_TL_UCP_KNOMIAL_RADIX with default value auto. If user sets some value for this variable then we force apply that value for all knomial collectives.

Reply:
added a convenience variable, plz have a look if that is what you meant.